### PR TITLE
DOCS-#1816: Add notes about using MODIN_SOCKS_PROXY variable

### LINKED ISSUE #1816: Add notes about using MODIN_SOCKS_PROXY variable
**Is your feature request related to a problem? Please describe.**
A clear and concise description of what the problem is. What kind of performance improvements would you like to see with this new API?

### PR DESCRIPTION
**Pull request by anmyachev**
Signed-off-by: Anatoly Myachev <anatoly.myachev@intel.com>

## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [x] passes `flake8 modin`
- [x] passes `black --check modin`
- [x] signed commit with `git commit -s` 
- [x] Resolves #1816 
- [ ] tests added and passing

### CODE REVIEW

**Comment by anmyachev** [comment]
> I don't think proxy belongs to this interface whatsoever.
> 
> Also requiring it to be hardcoded in the user files (and removing the ability to pass it through environment) is something I don't agree with.

The main advantage of my changes is that the user only needs to look at the function in order to understand what needs to be done for proxy environment. Now the user needs to look for the name of the variable in the source code. It also bad.

File: modin/experimental/cloud/base.py [hunk 1]
```diff
@@ -63,10 +64,10 @@ def _which(prog):
     return None
 
 
-def _get_ssh_proxy_command():
-    socks_proxy = os.environ.get("MODIN_SOCKS_PROXY", None)
+def _get_ssh_proxy_command(socks_proxy: str = None):
     if socks_proxy is None:
         return None
+
```
Comment (devin-petersohn) [inline modin/experimental/cloud/base.py:70]:
```suggestion
```

File: modin/experimental/cloud/cluster.py [hunk 2]
```diff
@@ -252,6 +252,11 @@ def create(
         The object that knows how to destroy the cluster and how to activate it as remote context.
         Note that by default spawning and destroying of the cluster happens in the background,
         as it's usually a rather lengthy process.
+
+    Notes
+    -----
+    Cluster computation actually can work in proxy environment.
+    You should setup "MODIN_SOCKS_PROXY" variable before call the function.
```
Comment (vnlitvinov) [inline modin/experimental/cloud/cluster.py:259]:
```suggestion
    Cluster computation actually can work  when proxies are required to access the cloud.
    You should set normal "http_proxy"/"https_proxy" variables for HTTP/HTTPS proxies and
    set "MODIN_SOCKS_PROXY" variable for SOCKS proxy before calling the function.
    
    Using SOCKS proxy requires Ray newer than 0.8.6, which might need to be installed manually.
```

**Comment by devin-petersohn** [comment]
I will let @vnlitvinov take another look

**Pull request merged**